### ISSUE TITLE
java.lang.RuntimeException: Error receiving broadcast Intent { act=android.intent.action.DOWNLOAD_COMPLETE

### ISSUE THREAD

**Issue by zileyuan** [4Q-s-r-o/ota_update#11]
```
java.lang.RuntimeException: Error receiving broadcast Intent { act=android.intent.action.DOWNLOAD_COMPLETE flg=0x10 pkg=com.ps.zhizuxia (has extras) } in sk.fourq.otaupdate.a@29239c0c
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:928)
    at android.os.Handler.handleCallback(Handler.java:739)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:134)
    at android.app.ActivityThread.main(ActivityThread.java:5377)
    at java.lang.reflect.Method.invoke(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:967)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:762)
Caused by: java.lang.NullPointerException: Attempt to invoke interface method 'void io.flutter.plugin.common.EventChannel$EventSink.success(java.lang.Object)' on a null object reference
    at sk.fourq.otaupdate.a.onReceive()
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:906)
    ... 8 more
java.lang.NullPointerException: Attempt to invoke interface method 'void io.flutter.plugin.common.EventChannel$EventSink.success(java.lang.Object)' on a null object reference
    at sk.fourq.otaupdate.a.onReceive()
    at android.app.LoadedApk$ReceiverDispatcher$Args.run(LoadedApk.java:906)
    at android.os.Handler.handleCallback(Handler.java:739)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:134)
    at android.app.ActivityThread.main(ActivityThread.java:5377)
    at java.lang.reflect.Method.invoke(Native Method)
    at java.lang.reflect.Method.invoke(Method.java:372)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:967)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:762)
```

**Comment by zileyuan** [comment]
I think in onReceive function throw error, possible progressSink is null
```
public void onReceive(Context c, Intent i) {
    //DOWNLOAD IS COMPLETE, UNREGISTER RECEIVER AND CLOSE PROGRESS SINK
    context.unregisterReceiver(this);
    //TRIGGER APK INSTALLATION
    Intent intent;
    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
        //AUTHORITY NEEDS TO BE THE SAME ALSO IN MANIFEST
        Uri apkUri = FileProvider.getUriForFile(context, "sk.fourq.ota_update.provider", new File(destination));
        intent = new Intent(Intent.ACTION_INSTALL_PACKAGE);
        intent.setData(apkUri);
        intent.setFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
    } else {
        intent = new Intent(Intent.ACTION_VIEW);
        intent.setDataAndType(fileUri, "application/vnd.android.package-archive");
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
    }
    //SEND INSTALLING EVENT
    progressSink.success(Arrays.asList("" + OtaStatus.INSTALLING.ordinal(), ""));
    progressSink.endOfStream();
    progressSink = null;
    context.startActivity(intent);
}
```

**Comment by AlexVincent525** [comment]
我们两个issue同样问题 😢 请问你解决了吗

**Comment by MartinHlavna** [comment]
Look like related to #10

**Comment by MartinHlavna** [comment]
Fix will be published in 2.0.1
